components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/RequestObjectValidatorUtil.java

```diff
@@ -0,0 +1,212 @@
+                                oAuth2Parameters.getClientId());
+                    }
+                    break;
+                }
+            }
+        } else {
+            return StringUtils.EMPTY;
+        }
+        return jwksUri;
+    }
+
+    /**
+     * Validating signature based on jwks endpoint.
+     *
+     * @param signedJWT signed JWT
+     * @param jwksUri   Uri of the JWKS endpoint
+     * @return signature validity
+     * @throws RequestObjectException
+     */
+    private static boolean isSignatureVerified(SignedJWT signedJWT, String jwksUri) throws RequestObjectException {
+
+        // Validate the signature of the assertion using the jwks endpoint.
+        if (StringUtils.isNotBlank(jwksUri)) {
+            String jwtString = signedJWT.getParsedString();
+            String alg = signedJWT.getHeader().getAlgorithm().getName();
+            try {
+                return new JWKSBasedJWTValidator().validateSignature(jwtString, jwksUri, alg, MapUtils.EMPTY_MAP);
+            } catch (IdentityOAuth2Exception e) {
+                String errorMessage = "Error occurred while validating request object signature using jwks endpoint";
+                throw new RequestObjectException(errorMessage, OAuth2ErrorCodes.SERVER_ERROR, e);
+            }
+        } else {
+            log.warn("JWKS URI is empty");
+        }
+        return false;
+
+    }
+
+    protected static Certificate getCertificateForAlias(String tenantDomain, String alias) throws
+            RequestObjectException {
+
+        return getX509CertOfOAuthApp(alias, tenantDomain);
+    }
+
+    protected static Certificate getX509CertOfOAuthApp(String clientId, String tenantDomain) throws
+            RequestObjectException {
+
+        try {
+            return OAuth2Util.getX509CertOfOAuthApp(clientId, tenantDomain);
+        } catch (IdentityOAuth2Exception e) {
+            String errorMsg = "Error retrieving application certificate of OAuth app with client_id: " + clientId +
+                    " , tenantDomain: " + tenantDomain;
+            if (StringUtils.isNotBlank(e.getMessage())) {
+                // We expect OAuth2Util.getX509CertOfOAuthApp() to throw an exception with a more specific reason for
+                // not being able to retrieve the X509 Cert of the service provider.
+                errorMsg = e.getMessage();
+            }
+            throw new RequestObjectException(errorMsg, e);
+        }
+    }
+
+    /**
+     * Validate the signedJWT signature with given certificate
+     *
+     * @param signedJWT signed JWT
+     * @param x509Certificate X509 certificate
+     * @return signature validity
+     */
+    private static boolean isSignatureVerified(SignedJWT signedJWT, Certificate x509Certificate) {
+
+        JWSVerifier verifier;
+        JWSHeader header = signedJWT.getHeader();
+        if (x509Certificate == null) {
+            return logAndReturnFalse("Unable to locate certificate for JWT " + header.toString());
```

Comment:
let's not have a separate method for this. check if debug enabled, then log with the concat, then return false. when passing to a seperate method string concat is already done and no point in checking for is debug enabled. 

Also it's not nice to merge these two which are two independent things

Reply:
Fixed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1743/commits/0d57e5040acf37b8685893a5c5cbd39f29ea9bed